src/platform/silabs/SilabsConfigCTM.cpp

```diff
@@ -0,0 +1,424 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    Copyright (c) 2019 Nest Labs, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Utilities for accessing persisted device configuration on
+ *          platforms based on the Silicon Labs SDK.
+ */
+#include <platform/silabs/SilabsConfig.h>
+
+#include <lib/core/CHIPEncoding.h>
+#include <lib/support/CodeUtils.h>
+#include <platform/internal/testing/ConfigUnitTest.h>
+#include <platform/silabs/CHIPDevicePlatformConfig.h>
+
+#include "uart.h"
+#include <FreeRTOS.h>
+#include <nvm3.h>
+#include <nvm3_default.h>
+#include <nvm3_hal_flash.h>
+#include <nvm3_lock.h>
+#include <semphr.h>
+#include <sl_token_manager_interface.h>
+// Substitute the GSDK weak nvm3_lockBegin and nvm3_lockEnd
+// for an application controlled re-entrance protection
+static SemaphoreHandle_t nvm3_Sem;
+static StaticSemaphore_t nvm3_SemStruct;
```

Comment:
Prefer anon namespace to static.
```suggestion
namespace {
SemaphoreHandle_t nvm3_Sem;
StaticSemaphore_t nvm3_SemStruct;
}
```